week-4-tests-and-persistence/w04e01-simple-testing/src/app/divider.spec.js

```diff
@@ -0,0 +1,25 @@
+import { divider } from "./divider";
+
+describe("divider", () => {
+  it("should divide two numbers", () => {
+    const result = divider(10, 2);
+    expect(result).toBe(5);
+  });
+
+  it("should return 0 if dividend is 0", () => {
+    const result = divider(0, 2);
+    expect(result).toBe(0);
+  });
+
+  it("should throw an error when divisor is not number", () => {
+    expect(() => divider(10, "a")).toThrowError();
+  });
+
+  it("should throw an error when dividend is not number", () => {
+    expect(() => divider("a", 10)).toThrowError();
+  });
+
+  it("should throw an error when divisor is 0", () => {
+    expect(() => divider(10, 0)).toThrowError();
+  });
```

Comment:
Bardzo ładnie określone przypadki testowe i wpadłeś na miejsca - o które mi chodziło żeby przetestować.

Widzę że podobają Ci się _Github Actions_ - też fajnie wpięte :).

Jedyny przypadek który mi przychodzi luźno do głowy, to liczby zmiennoprzecinkowe żeby dorzucić do testowania.
Też jakiś prosty test żeby to potwierdzić.

Dodatkowo zobacz sobie coś takiego, czasem przydatne (tylko nie przesadzaj z ilością przypadków testowych :)

https://vitest.dev/api/#test-each

```javascript
it.each( [ 
[] 
] )(() => /*...*/)
```

Reply:
Co można automatyzować to trzeba automatyzować 😎

Reply:
Ok jak git, to możesz zamknąć 👍